packages/lib-core/src/yup-schemas.ts

```diff
@@ -41,13 +41,34 @@ const extensionTypeSchema = yup
   .required()
   .matches(/^[a-z]+[a-z-]*\.[a-z]+[a-z-]*(?:\/[a-z]+[a-z-]*)*$/, 'extension type');
 
+/**
+ * Schema for a valid feature flag name.
+ *
+ * Examples:
+ *
+ * ```
+ * FOO, FOO_BAR
+ * ```
+ */
+const featureFlagNameSchema = yup
+  .string()
+  .required()
+  .matches(/^[A-Z]+[A-Z_]*$/, 'feature flag name');
```

Comment:
Do we not allow numbers in the flag name?

Reply:
Sounds reasonable, let's allow numbers as well

```diff
-  .matches(/^[A-Z]+[A-Z_]*$/, 'feature flag name');
+  .matches(/^[A-Z]+[A-Z0-9_]*$/, 'feature flag name');
```